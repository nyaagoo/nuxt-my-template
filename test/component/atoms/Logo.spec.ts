import { createLocalVue, mount, shallowMount } from '@vue/test-utils';
import Vuex, { Store } from 'vuex';
import {
  initializeStores,
  dogStore,
  counterStore
} from '~/utils/store-accessor';
import Dog from '~/store/dog';
import Counter from '~/store/counter';
import Logo from '~/components/atoms/Logo.vue';

const localVue = createLocalVue();
localVue.use(Vuex);
let store: Store<any>;
describe('Logo', () => {
  beforeAll(() => {
    store = new Vuex.Store({
      modules: { dog: Dog, counter: Counter }
    });
    initializeStores(store);
  });
  test('Logo.vue can mount', () => {
    const wrapper = mount(Logo);
    expect(wrapper.isVueInstance).toBeTruthy();
  });

  test('Snapshot', () => {
    const wrapper = shallowMount(Logo);
    expect(wrapper.element).toMatchSnapshot();
  });

  test('Module can access another module function', () => {
    dogStore.addDog({ name: 'sa', ruby: 'sa' });
    expect(counterStore.counter1).toBe(11);
  });
});
