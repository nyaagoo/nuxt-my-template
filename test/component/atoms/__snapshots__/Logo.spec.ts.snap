// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Snapshot 1`] = `
<div
  class="VueToNuxtLogo"
>
  <div
    class="Triangle Triangle--two"
  />
  <div
    class="Triangle Triangle--one"
  />
  <div
    class="Triangle Triangle--three"
  />
  <div
    class="Triangle Triangle--four"
  />
</div>
`;
